src/EventBuilder/Cecil/EventTemplateInformation.cs

```diff
@@ -81,6 +79,20 @@ private static string GetRealTypeName(TypeReference t)
             return ret.Replace('/', '.');
         }
 
+        private static ObsoleteEventInfo GetObsoleteInfo(EventDefinition ei)
+        {
+            var obsoleteAttribute = ei.CustomAttributes.FirstOrDefault(attr => attr.AttributeType.FullName.Equals("System.ObsoleteAttribute"));
+
+            if (obsoleteAttribute == null) 
+                return null;
+
+            return new ObsoleteEventInfo
+            {
+                Message = obsoleteAttribute?.ConstructorArguments?.ElementAtOrDefault(0).Value?.ToString() ?? string.Empty,
```

Comment:
If `ElementAtOrDefault(0)` is indeed the default (`null`), won't this crash? Shouldn't it just be using `ElementAt`, since we'd expect that ctor argument to _always_ be present? Same goes for line below.

Reply:
I tested it manually with prepared example library:
```csharp
 public class Class1
    {
        public event EventHandler SampleEvent;

        [Obsolete()]
        public event EventHandler SampleObsoleteEvent;

        [Obsolete("Some message")]
        public event EventHandler SampleObsoleteEventWithMessage;

        [Obsolete("Some message", true)]
        public event EventHandler SampleObsoleteEventWithMessageAndError;
    }
```
The results is:
```csharp
public IObservable<System.EventArgs> SampleEvent {
            get { return Observable.FromEventPattern<System.EventHandler, System.EventArgs>(x => This.SampleEvent += x, x => This.SampleEvent -= x).Select(x => x.EventArgs); }
        }


        [Obsolete]
        public IObservable<System.EventArgs> SampleObsoleteEvent {
            get { return Observable.FromEventPattern<System.EventHandler, System.EventArgs>(x => This.SampleObsoleteEvent += x, x => This.SampleObsoleteEvent -= x).Select(x => x.EventArgs); }
        }


        [Obsolete("Some message")]
        public IObservable<System.EventArgs> SampleObsoleteEventWithMessage {
            get { return Observable.FromEventPattern<System.EventHandler, System.EventArgs>(x => This.SampleObsoleteEventWithMessage += x, x => This.SampleObsoleteEventWithMessage -= x).Select(x => x.EventArgs); }
        }


        [Obsolete("Some message", true)]
        public IObservable<System.EventArgs> SampleObsoleteEventWithMessageAndError {
            get { return Observable.FromEventPattern<System.EventHandler, System.EventArgs>(x => This.SampleObsoleteEventWithMessageAndError += x, x => This.SampleObsoleteEventWithMessageAndError -= x).Select(x => x.EventArgs); }
        }
```

I intentionally used null-conditional operator to avoid such problems.

Reply:
It's not using the elvis operator though for the case I'm talking about:
```
ElementAtOrDefault(0).Value
```

Reply:
`ElementAt()` throws exception in case element does not exist. You can use `[Obsolete]` attribute without any arguments. As I see we can remove one null check for `ConstructorArguments` - it's always initialized. So it'll be: `obsoleteAttribute.ConstructorArguments.ElementAtOrDefault()`

Reply:
I think you might be missing my point :)

I realize `ElementAt` throws when the element doesn't exist. But so does `ElementAtOrDefault(0).Value`. The difference is that `ElementAt` will throw with an `IndexOutOfRangeException` whereas the latter will throw a less useful `NullReferenceException`.

So my question is around how this code actually works when there is no element at index `0`, for example. I'm suggesting it will crash with NRE because `Value` is being dereferenced.

Reply:
It works because `CustomAttributeArgument` is a struct (value type).

Reply:
Aha! Perfect - thanks, I obviously did not realise that. Sorry for wasting your time :)

Reply:
No problem :)